website/www/site/content/en/documentation/sdks/java/testing/tpcds.md

```diff
@@ -0,0 +1,183 @@
+to this, the necessary dependencies must be configured.
+
+When running via Gradle, the following two parameters control the execution:
+
+    -P tpcds.args
+        The command line to pass to the TPC-DS main program.
+
+    -P tpcds.runner
+	The Gradle project name of the runner, such as ":runners:spark:3" or
+	":runners:flink:1.13. The project names can be found in the root
+        `settings.gradle.kts`.
+
+Test data has to be generated before running a suite and stored to accessible file system. The query results will be written into output files.
+
+### Common configuration parameters
+
+Size of input dataset (1GB / 10GB / 100GB / 1000GB):
+
+    --dataSize=<1GB|10GB|100GB|1000GB>
+
+Path to input datasets directory:
+
+    --dataDirectory=<path to dir>
+
+Path to results directory:
+
+    --resultsDirectory=<path to dir>
+
+Format of input files:
+
+    --sourceType=<CSV|PARQUET>
+
+Run queries (comma separated list of query numbers or `all` for all queries):
+
+    --queries=<1,2,...N|all>
+
+Number of queries **N** that are running in parallel:
+
+    --tpcParallel=N
+
+## Running TPC-DS
+
+There are some examples how to run TPC-DS benchmark on different runners.
+
+Running suite on the SparkRunner (local) with Query3 against 1Gb dataset in Parquet format:
+
+    ./gradlew :sdks:java:testing:tpcds:run \
+        -Ptpcds.runner=":runners:spark:3" \
+        -Ptpcds.args="
+            --runner=SparkRunner
+            --dataSize=1GB
+            --sourceType=PARQUET
+            --dataDirectory=gs://beam-tpcds/datasets/parquet/partitioned
+            --resultsDirectory=/tmp/beam-tpcds/results/spark/
+            --tpcParallel=1
+            --queries=3"
+
+Running suite on the FlinkRunner (local) with Query7 and Query10 in parallel against 10Gb dataset in CSV format:
+
+    ./gradlew :sdks:java:testing:tpcds:run \
+        -Ptpcds.runner=":runners:flink:1.13" \
+        -Ptpcds.args="
+            --runner=FlinkRunner
+            --parallelism=2
+            --dataSize=10GB
+            --sourceType=CSV
+            --dataDirectory=gs://beam-tpcds/datasets/csv
+            --resultsDirectory=/tmp/beam-tpcds/results/flink/
+            --tpcParallel=2
+            --queries=7,10"
+
+Running suite on the DataflowRunner (local) with all queries against 100Gb dataset in PARQUET format:
+
+    ./gradlew :sdks:java:testing:tpcds:run \
+        -Ptpcds.runner=":runners:google-cloud-dataflow-java" \
+        -Ptpcds.args="
+            --runner=DataflowRunner
+            --region=<region_name>
+            --project=<project_name>
+            --numWorkers=4
+            --maxNumWorkers=4
+            --autoscalingAlgorithm=NONE
+            --dataSize=100GB
+            --sourceType=PARQUET
+            --dataDirectory=gs://beam-tpcds/datasets/parquet/partitioned
+            --resultsDirectory=/tmp/beam-tpcds/results/dataflow/
+            --tpcParallel=4
+            --queries=all"
+
+## TPC-DS dashboards
+TBD
```

Comment:
I don't think we should publish this with the "TBD" sections still here, do you intend to address these? If not we could just comment them out for now so they don't show up on the website.

Reply:
Commented for now. I will add this section once it will be done in Grafana